core/services/pg/q.go

```diff
@@ -179,5 +210,51 @@ func (q Q) GetNamed(sql string, dest interface{}, arg interface{}) error {
 	}
 	ctx, cancel := q.Context()
 	defer cancel()
-	return errors.Wrap(q.GetContext(ctx, dest, query, args...), "error in get query")
+
+	q.logSqlQuery(query, args...)
+	begin := time.Now()
+	defer q.postSqlLog(ctx, begin)
+
+	return q.withLogError(errors.Wrap(q.GetContext(ctx, dest, query, args...), "error in get query"))
+}
+
+type queryFmt struct {
+	query string
+	args  []interface{}
+}
+
+func (q queryFmt) String() string {
+	if q.args == nil {
+		return q.query
+	}
+	var pairs []string
+	for i, arg := range q.args {
+		pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("%v", arg))
+	}
+	replacer := strings.NewReplacer(pairs...)
+	return replacer.Replace(q.query)
+}
+
+func (q Q) logSqlQuery(query string, args ...interface{}) {
+	if q.logger == nil || q.config == nil || !q.config.LogSQLStatements() {
+		return
+	}
+	q.logger.Debugf("SQL: %s", queryFmt{query, args})
+}
+
+func (q Q) withLogError(err error) error {
```

Comment:
One thing to note here: this will log `sql.ErrNoRows` which is often expected and can get quite annoying.

I would be ok with making `withLogError` only log when `LOG_SQL` is enabled.

I think it still makes sense to always log slow queries.